Modules/_collectionsmodule.c

```diff
@@ -1077,37 +1106,54 @@ deque_index(dequeobject *deque, PyObject *const *args, Py_ssize_t nargs)
         start = stop;
     assert(0 <= start && start <= stop && stop <= Py_SIZE(deque));
 
-    /* XXX Replace this loop with faster code from deque_item() */
-    for (i=0 ; i<start ; i++) {
-        index++;
-        if (index == BLOCKLEN) {
-            b = b->rightlink;
-            index = 0;
-        }
-    }
+    index = start;
+    b = deque_get_block(deque, &index);
 
-    n = stop - i;
+    n = stop - start ;
     while (--n >= 0) {
         CHECK_NOT_END(b);
         item = b->data[index];
-        cmp = PyObject_RichCompareBool(item, v, Py_EQ);
+        cmp = PyObject_RichCompareBool(item, value, Py_EQ);
         if (cmp > 0)
-            return PyLong_FromSsize_t(stop - n - 1);
+            return stop - n - 1;
         if (cmp < 0)
-            return NULL;
+            return -1;
         if (start_state != deque->state) {
             PyErr_SetString(PyExc_RuntimeError,
                             "deque mutated during iteration");
-            return NULL;
-        }
-        index++;
-        if (index == BLOCKLEN) {
-            b = b->rightlink;
-            index = 0;
+            return -1;
         }
+
+        do {
```

Comment:
Why is the `do { ... } while (0)` wrapping needed here?

Reply:
This was a macro before and I forgot to remove the wrapping. Sorry about that. Corrected in 6317c78